src/Orleans/Core/GrainFactory.cs

```diff
@@ -290,6 +405,35 @@ internal static TGrainInterface GetSystemTarget<TGrainInterface>(GrainId grainId
         #endregion
 
         #region Utility functions
+
+        /// <summary>
+        /// Ensures that generated code reflects the current runtime environment.
+        /// </summary>
+        private static void EnsureCodeGenerationIsCurrent()
+        {
+            // Get the code generator if it is available.
+            if (codeGenerator == null)
```

Comment:
If `codeGen` is unexpectedly null, shouldn't we log and/or throw instead of silently skipping codegen?